### ISSUE TITLE
Visual Studio reports warnings in auto-generated tagged files

### ISSUE THREAD

**Issue by IXLLEGACYIXL** [dotnet/roslyn#70280]
**Version Used**: 
netstandard2.0 for the Source Generator
net6.0-windows for the project that receives the generations

Visual Studio
Microsoft Visual Studio Community 2022 (64-Bit) - Current
Version 17.7.3
**Steps to Reproduce**:

1. Generate a file with roslyn which has nullable enable and auto-generated comment at the top and .g.cs in the name
2. Create a method with an unused variable
3. VS still complains about the not used variable
4. Also you can create a case for possible null assignment `string AssemblyName { get; } = typeof(whatever).Assembly.GetName().Name;`

**Diagnostic Id**:

multiple... all nullable diagnostics and not assigned diagnostics
`CS8601` and `CS8766` i can prove in the images.

**Expected Behavior**:
VS should not complain about warnings in auto generated context

**Actual Behavior**:
it complains

**My Case Shown**
```cs
// <auto-generated/>
//  This code was generated by Strides YamlSerializer.
//  Do not edit this file.

#nullable enable
using System;
using System.Linq;
using System.Collections.Generic;
using VYaml;
using VYaml.Emitter;
using VYaml.Parser;
using VYaml.Serialization;
using Stride.Core;
namespace Core;
[System.CodeDom.Compiler.GeneratedCode("StrideYaml","1.0.0.0")]
file class StrideSourceGenerated_CoreTInherit : IYamlFormatter<TInherit>
{
    string AssemblyName { get; } = typeof(TInherit).Assembly.GetName().Name;
    string IdentifierTag { get; } = typeof(TInherit).Name;
    Type IdentifierType { get; } = typeof(TInherit);

    public void Register()
    {
        NexYamlSerializerRegistry.Instance.RegisterFormatter(this);
    }

    public void Serialize(ref Utf8YamlEmitter emitter, TInherit value, YamlSerializationContext context)
    {
        var temp_Nun = default(System.Int32);

    }

    public TInherit? Deserialize(ref YamlParser parser, YamlDeserializationContext context)
    {
        return default!;
    }
}
```

this results in VS in this diagnostics
![grafik](https://github.com/dotnet/roslyn/assets/107197024/c2ef82eb-7580-432d-9f4d-518914780887)
![grafik](https://github.com/dotnet/roslyn/assets/107197024/7a5ea8f5-4512-4f3f-9eaf-d6a28bb3a6c0)
![grafik](https://github.com/dotnet/roslyn/assets/107197024/a9cf88ad-e46e-46f4-a8a9-aef69e70012f)

even though it shouldnt complain at all, ive added the tags of .g.cs and auto generated comment on the script
there is no empty line above the auto generated comment

**Comment by Youssef1313** [comment]
This is looking by-design. You explicitly added `#nullable enable` to the generated code, so nullable analysis is performed and produces warnings. Either you need to remove `#nullable enable` from the source generator, or make the source generator produce code that satisfies the compiler analysis.

**Comment by IXLLEGACYIXL** [comment]
allright, i thought wenn a file has one of the source generation tags ( .g.cs or auto-generated ) that  VS doesnt scan it anymore
